plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/WlmModeManager.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.wlm.WlmMode;
+import org.opensearch.wlm.WorkloadManagementSettings;
+
+/**
+ * Central manager for maintaining the current WLM mode with updates from cluster settings.
+ */
+public class WlmModeManager {
```

Comment:
This class should be renamed to something which doesn't tie it to a single non plugin setting.
for example `NotPluginSettingValuesProvider`